x-pack/plugin/core/src/main/java/module-info.java

```diff
@@ -23,6 +23,8 @@
 
     exports org.elasticsearch.index.engine.frozen;
     exports org.elasticsearch.license;
+    exports org.elasticsearch.license.internal to org.elasticsearch.stateless;
+    opens org.elasticsearch.license.internal; //spi
```

Comment:
I would expect to be able to qualify the opens such as: 
```
opens org.elasticsearch.license.internal to org.elasticsearch.server; //spi
```
However, when testing with that qualification (and a dozen other variants) it fails with the following stack trace and a message [1] that does not make any sense (see the line above). It would be nice to qualify this opens, but I can't figure out the magic to get this to work and it's probably not a an issue since the export is qualified. 

[1]
```java
java.lang.IllegalAccessError: superinterface check failed: class co.elastic.elasticsearch.stateless.license.ServerlessLicense (in module org.elasticsearch.stateless) cannot access class org.elasticsearch.license.internal.MutableLicenseService (in module org.elasticsearch.xcore) because module org.elasticsearch.xcore does not export org.elasticsearch.license.internal to module org.elasticsearch.stateless
  at java.base/java.lang.ClassLoader.defineClass1(Native Method)
  at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1013)
  at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1091)
  at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:182)
  at java.base/jdk.internal.loader.Loader.defineClass(Loader.java:612)
  at java.base/jdk.internal.loader.Loader.lambda$findClassInModuleOrNull$6(Loader.java:589)
  at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
  at java.base/jdk.internal.loader.Loader.findClassInModuleOrNull(Loader.java:590)
  at java.base/jdk.internal.loader.Loader.loadClass(Loader.java:548)
  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
  at java.base/java.lang.Class.forName0(Native Method)
  at java.base/java.lang.Class.forName(Class.java:496)
  at java.base/java.lang.Class.forName(Class.java:475)
  at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.plugins.spi.SPIClassIterator.next(SPIClassIterator.java:161)
  at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.plugins.PluginsService.createExtensions(PluginsService.java:353)
  at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.plugins.PluginsService$1.loadExtensions(PluginsService.java:340)
  at org.elasticsearch.xcore@8.8.0-SNAPSHOT/org.elasticsearch.xpack.core.XPackPlugin.loadExtensions(XPackPlugin.java:488)
  at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.plugins.PluginsService.loadExtensionsForPlugin(PluginsService.java:346)
  at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.plugins.PluginsService.loadExtensions(PluginsService.java:304)
  at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.plugins.PluginsService.loadBundles(PluginsService.java:293)
  at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:165)
  at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.plugins.PluginsService.lambda$getPluginsServiceCtor$16(PluginsService.java:666)
  at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.node.Node.<init>(Node.java:419)
  at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.node.Node.<init>(Node.java:326)
  at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.bootstrap.Elasticsearch$2.<init>(Elasticsearch.java:216)
  at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.bootstrap.Elasticsearch.initPhase3(Elasticsearch.java:216)
  at org.elasticsearch.server@8.8.0-SNAPSHOT/org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:67)
```